lib/subjects/designator_selector.rb

```diff
@@ -34,7 +34,12 @@ def get_subjects(user, group_id, limit)
       return [] unless enabled?
 
       subject_ids = self.class.client.get_subjects(workflow.id, user.try(&:id), group_id, limit)
+      return [] unless subject_ids.present?
+      raise "Hacking attempt" unless subject_ids.all? { |i| i.is_a? Integer }
+
       sms_scope = SetMemberSubject.by_subject_workflow(subject_ids, workflow.id)
+        .order("idx(array[#{subject_ids.join(',')}], set_member_subjects.subject_id)")
```

Comment:
I've was thinking about this the other day, what do you say we can just return the subject ids from the selector as they? The selectors check the set linkage when loading data and keep things in sync under normal operations, why do we need this extra db query to check the set linkage? I think this is super vestigial and we can optimize the selection path by removing this extra sms scope selection. Thoughts?

Reply:
Yeah, I'm up for removing it. I'll look at it next week.